### PR TITLE
Update kubeadm's minimum supported Kubernetes in v1.11.x to 1.10

### LINKED ISSUE #815: Bump minimum version to v1.10 and cleanup v1.9-specific code at HEAD
Same as https://github.com/kubernetes/kubeadm/issues/622. Very important to get merged in v1.11.

 - Bump minimum supported k8s and kubelet version in constants.go
 - Bump etcd version based on k8s version to 1.10.X => 3.1.12, 1.11.X => 3.2.18, 1.12.X => 3.2.18
 - Remove `var v190` from `features.go`
 - Rename `v180AndAboveKubeDNSDeployment` to `KubeDNSDeployment` in `phases/addons/dns/versions.go`
 - Remove `v190alpha3` from `postupgrade.go`

The need for this is explained in https://github.com/kubernetes/kubeadm/blob/master/docs/release-cycle.md

/assign @dixudx

### PR DESCRIPTION
**Pull request by dixudx**
**What this PR does / why we need it**:
This PR does,

* Restricts supported Kubernetes version in 1.11 cycle;
* Removes useless v190-specific variables, since the minimum version is v1.10;
* Bumps etcd version based on k8s version to 1.10.X => 3.1.12, 1.11.X => 3.2.18, 1.12.X => 3.2.18;

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes kubernetes/kubeadm#815

**Special notes for your reviewer**:
/cc kubernetes/sig-cluster-lifecycle-pr-reviews
/cc luxas 
**Release note**:

```release-note
Update kubeadm's minimum supported kubernetes in v1.11.x to 1.10
```

### CODE REVIEW

**Comment by luxas** [comment]
Thanks! Please do these as well:
> Bump etcd version based on k8s version to 1.10.X => 3.1.12, 1.11.X => 3.2.18, 1.12.X => 3.2.18
> Remove var v190 from features.go
> Rename v180AndAboveKubeDNSDeployment to KubeDNSDeployment in phases/addons/dns/versions.go
> Remove v190alpha3 from postupgrade.go

**Comment by dixudx** [comment]
@luxas I've split them into separate PRs for better review and rebasing.

**Review by luxas — APPROVED** [review]
/lgtm
/approve

Thanks a lot @dixudx :clap:!

File: cmd/kubeadm/app/phases/upgrade/compute.go [hunk 1]
```diff
@@ -279,7 +279,7 @@ func GetAvailableUpgrades(versionGetterImpl VersionGetter, experimentalUpgradesA
 }
 
 func getBranchFromVersion(version string) string {
-	return strings.TrimPrefix(version, "v")[:3]
```
Comment (luxas) [inline cmd/kubeadm/app/phases/upgrade/compute.go]: I see this works, but I'd like to make it more resilient by using `versionutil` package for parsing and then invoking `v.Minor()`. Can you follow-up a PR with that please?

**Review by luxas — APPROVED** [review]
/lgtm

Thanks!

File: cmd/kubeadm/app/phases/upgrade/compute.go (hunk 1 above)
Comment (dixudx) [reply cmd/kubeadm/app/phases/upgrade/compute.go]: Yeah. Addressed this in #64008.

**Pull request merged**